web/modules/custom/joinup_news/config/install/joinup_news.settings.yml

```diff
@@ -0,0 +1,110 @@
+# The array of allowed conditions for news workflow transitions.
+# Format of array:
+# @code
+# [
+#   <parent moderation> => [
+#     <target state> => [
+#       <source state> => [
+#         <role1>
+#         <role2>
+#         .
+#         .
+#       ]
+#     ]
+#   ]
+# ]
+# @endcode
+#
+# The array is a four dimensions array. The first level contains the
+# value of the moderation of the parent entity. The second level is the
+# target state which is tha state that the entity transits to. Each target
+# state is an array of allowed source states which in every check, is the
+# current state of the entity. Finally, the source states are arrays
+# of roles that are allowed to perform this action.
```

Comment:
Phrasing can be reworked a bit here, maybe.

> ... the value of the moderation of the parent entity ...

could be

> ... the moderation value of the parent entity ... 

Again here
> The second level is the state which is tha state that the entity transits to

could be 

> The second level is/represents the target state the entity transits to.

This one I think need to be rewritten, as it's not clear.

> Each target is an array of allowed source states which in every check, is the state of the entity.